### ISSUE TITLE
GraphQL request : TransactionChain pagination

### ISSUE THREAD

**Issue by redDwarf03** [archethic-foundation/archethic-node#296]
### What happened?

I created 14 transactions in a transaction chain

when I use a request with ```page: 0```, I obtain the 4 last transactions
```query { transactionChain(address: "00003C85EB8CC60B73A049FE80513BB5DD2204B6A506DD8D83ACC34FEE40B21CAA94", page: 0) {  address} }

{
  "data": {
    "transactionChain": [
      {
        "address": "00000C37AA448DB2913EC62E232C27BFB096D09542457778FB2AD04B0B56F8C18092"
      },
      {
        "address": "00000D597B72BD3711588A48504698F389E40B144F16BD94B8449E160AB51A7B731C"
      },
      {
        "address": "00008CE5F15A7564AB73F1589BE1A74A6A5F993DC377CFDBDD9FFADD40E5D1EAAAC4"
      },
      {
        "address": "00003C85EB8CC60B73A049FE80513BB5DD2204B6A506DD8D83ACC34FEE40B21CAA94"
      }
    ]
  }
}
```

when I use a request with ```page: 1```, I obtain the 10 first transactions I created
```query { transactionChain(address: "00003C85EB8CC60B73A049FE80513BB5DD2204B6A506DD8D83ACC34FEE40B21CAA94", page: 0) {  address} }

{
  "data": {
    "transactionChain": [
      {
        "address": "00007C7FAC9A741BF522EAF69824945B0721155C307270AE843A55F42C63F4526EC9"
      },
      {
        "address": "00000A9CE7F2DE51CFA0D3130099B2C6787AEB866647FD84945A124B21B14BB09D0E"
      },
      {
        "address": "00009D1AEA08BD7AF11DBEA0CA8CEAC2DA7C28B6CCEE4C69576827DB1C497655E080"
      },
      {
        "address": "000019565DF24A9AA5F82612E93BDB6BE18BCBA1158E74DBA810543C4948B523B79B"
      },
      {
        "address": "0000E15BF01140218B6A6269A514B8DCB5EA395A1C3351976EBE4FA2FCBFFC061917"
      },
      {
        "address": "0000FBB80E7223372C82C2CADF7C5A979AF1F266D1F404EBF5D3B231BFB0A3CD8CB0"
      },
      {
        "address": "0000E552B131BEEEBFE89903C9A97BD72424BCE755DD1BDFCDD9F31F127C6520B050"
      },
      {
        "address": "0000F7DA1CFD49A27AE101949D75F146A8C028EDD5A47B1DADD6EBEA373F1056C299"
      },
      {
        "address": "0000174CF96C37CD6A1020A367F75E1EC0193CC25EC2ABA909EAA185B25A652F0EE0"
      },
      {
        "address": "00000DF8F827A6DEC3C00466B644DA8F4DA04B608F93CD1BD20F7F3408158FF08510"
      }
    ]
  }
}
```

But, when I use a request with ```page: 2```, normally, I should have nothing (I think) but I obtain AGAIN the 4 last transactions
```
query { transactionChain(address: "00003C85EB8CC60B73A049FE80513BB5DD2204B6A506DD8D83ACC34FEE40B21CAA94", page: 2) {  address} }

{
  "data": {
    "transactionChain": [
      {
        "address": "00000C37AA448DB2913EC62E232C27BFB096D09542457778FB2AD04B0B56F8C18092"
      },
      {
        "address": "00000D597B72BD3711588A48504698F389E40B144F16BD94B8449E160AB51A7B731C"
      },
      {
        "address": "00008CE5F15A7564AB73F1589BE1A74A6A5F993DC377CFDBDD9FFADD40E5D1EAAAC4"
      },
      {
        "address": "00003C85EB8CC60B73A049FE80513BB5DD2204B6A506DD8D83ACC34FEE40B21CAA94"
      }
    ]
  }
}
```

and finally, with ```page 3```, I obtain
```{
  "data": {
    "transactionChain": null
  }
}
```

why I obtain same result in page 0 and page 2 ?

thx
 

### What is expected ?

could you precise me if it's a bug or If I don"t use correctly "page" property.

Other thing: could you tell me where I can't obtain the number of the transaction in the chain. Should I do calculate with the number of page * 10  + number of the transaction in the page ?

**Comment by samuel-uniris** [comment]
I think the issue is the page number validation.
So the implementation works a bit like this: `Enum.at(transactions_split_by_page, page - 1)`
But in Elixir `Enum.at(list, -1)` when you are using negative index is start to fetch from the end, so in your case is equal to take the last element of the list.
Then, because transactions_split_by_page in your case is: 
```
[
   page1,
   page2
]
```
Using page: 0 will mean page: 2

So thanks for the report, we have to change to add a better parameter validation to support only positive number greater than 1

**Comment by samuel-uniris** [comment]
But to avoid loading the entire chain we might change parameter to not use a number but the pagination state (last address retrieved) returned from the db.
So we will create new issue for this one.

**Comment by samuel-uniris** [comment]
Closed by #314
